lib/yard/parser/ruby/ruby_parser.rb

```diff
@@ -613,7 +620,7 @@ def comment_starts_line?(charno)
 
         def insert_comments
           root.traverse do |node|
-            next if [:comment, :void_stmt, :list].include?(node.type) || node.parent.type != :list
+            next if COMMENT_SKIP_NODE_TYPES.include?(node.type) || node.parent.type != :list
```

Comment:
Because of how Set is implemented in MRI, the array implementation is likely to be faster at this scale. (`set` is also not a prior dependency of `yard`, although that is moot as of Ruby 3.2)